### PR TITLE
Feature/5555 add timm support

### PR DESCRIPTION
**Pull request by potipot**
## What does this PR do?

Fixes # (https://github.com/PyTorchLightning/pytorch-lightning/issues/5555) <- introduce support for timm scheduler and optimizer. Waiting for someone else I decided to try and do it myself. This is my first PR so looking forward for your comments.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [ ] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@potipot we have locked master as there won't be any more releases in 1.1.x
pls see https://pytorch-lightning.readthedocs.io/en/latest/CONTRIBUTING.html#how-to-fix-pr-with-mixed-base-and-target-branches

File: pytorch_lightning/trainer/connectors/optimizer_connector.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 # limitations under the License.
 from pytorch_lightning.utilities import rank_zero_warn
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from timm.scheduler.scheduler import Scheduler as TimmScheduler
```
Comment (SkafteNicki) [inline pytorch_lightning/trainer/connectors/optimizer_connector.py:16]: As `timm` is an extra feature, we need to make sure that the code can run even if the user does not have timm installed. Something like this should be done:
https://github.com/PyTorchLightning/pytorch-lightning/blob/0a50bb406fa41dfa6a0e2be52f531a9c81c87d00/pytorch_lightning/loggers/wandb.py#L30-L37

File: pytorch_lightning/trainer/optimizers.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
 from pytorch_lightning.core.optimizer import LightningOptimizer
 from pytorch_lightning.utilities import rank_zero_warn
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
+import timm
```
Comment (SkafteNicki) [inline pytorch_lightning/trainer/optimizers.py:27]: Same as above comment

**Review by SkafteNicki — COMMENTED** [review]
I am not that familiar with the `timm` library but I can see that it seems to be quite popular, so I think we should support it :]
Some other comments:
* this function needs to be updated:
https://github.com/PyTorchLightning/pytorch-lightning/blob/7f352cb69a8202e3f829419657597697ca5d99e2/pytorch_lightning/trainer/optimizers.py#L144
* the documentation needs to be updated:
https://github.com/PyTorchLightning/pytorch-lightning/blob/release/1.2-dev/docs/source/common/optimizers.rst
* tests...

**Comment by jinwon-samsung** [comment]
How is this request going?

**Pull request closed without merging**